plans/network/pingpong.go

```diff
@@ -43,23 +51,25 @@ func pingpong(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	runenv.RecordMessage("before netclient.MustConfigureNetwork")
 	netclient.MustConfigureNetwork(ctx, config)
 
-	seq := client.MustSignalAndWait(ctx, "ip-allocation", runenv.TestInstanceCount)
-
 	// Make sure that the IP addresses don't change unless we request it.
 	if newAddrs, err := net.InterfaceAddrs(); err != nil {
 		return err
 	} else if !sameAddrs(oldAddrs, newAddrs) {
 		return fmt.Errorf("interfaces changed")
 	}
 
+	seq := client.MustSignalAndWait(ctx, "ip-allocation", runenv.TestInstanceCount)
```

Comment:
To me, it makes more sense to first check if the network interfaces have changed, before moving onto the next logical step (acquiring a sequence number).

Reply:
Do you know what could cause the addresses to change? I just want to make sure it's not something that can occur during wait for `ip-allocation`.

Reply:
The check refers to line 52, and possible side-effects:
`netclient.MustConfigureNetwork(ctx, config)`
Should configure the network according to the specs in the `config` struct, but it should not change the data IP address, because a new address is not specified in the config. Therefore, if the addresses change, the network config func has most likely encountered an error and the test should stop.

The `ip-allocation` wait is handled by the sync-service, and does not change the network interfaces of the test instance; that's why it made sense for me to switch the order of the code.

Reply:
Cool! Thanks for the explanation :)